app/src/org/commcare/xml/AndroidTransactionParserFactory.java

```diff
@@ -87,7 +88,13 @@ public void initCaseParser() {
             @Override
             public TransactionParser<Case> getParser(KXmlParser parser) {
                 if (created == null) {
-                    created = new AndroidCaseXmlParser(parser, tallies, true, sandbox.getCaseStorage(), generator);
+                    created = new AndroidCaseXmlParser(parser, tallies, true, sandbox.getCaseStorage(), generator) {
+
+                        @Override
+                        public void onIndexDisrupted(String caseId) {

```

Comment:
The regex that detects cross requests is brittle, only recognizing `cross-request: link`. Feel free to extend it: https://github.com/dimagi/mobile-deploy/blob/master/checkout_cross_request_repo.py#L22